kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -662,18 +662,16 @@ public List<DomainMetadata> getComputedDomainMetadatasToCommit() {
       }
 
       generateClusteringDomainMetadataIfNeeded();
-
-      if (domainsToAdd.isEmpty() && domainsToRemove.isEmpty()) {
-        // If no domain metadatas are added or removed, return an empty list. This is to avoid
-        // unnecessary loading of the domain metadatas from the snapshot (which is an expensive
-        // operation).
-        computedMetadatas = Optional.of(Collections.emptyList());
-        return Collections.emptyList();
-      }
-
       // Add all domains added in the transaction
       List<DomainMetadata> result = new ArrayList<>(domainsToAdd.values());
 
+      if (domainsToRemove.isEmpty()) {
```

Comment:
Q: why did we have to move this down a few lines from above? The only thing that we have changed is we have created a `result` variable -- but this is just equal to `domainToAdd.values()`

Reply:
Just no need to convert `domainToAdd.values()` to list twice